### PR TITLE
Fixes some errors with make EXTRA_ARGS="--language {langId}"

### LINKED ISSUE #3958: [BUG]: `make dev EXTRA_ARGS="--language {langId}"` is broken
### Describe the bug

I just tried to do `make dev EXTRA_ARGS="--lang ispc"` and the compiler pane shows no compiler, even though there is one compiler available.

Tracked it down to `Compiler` getting `state.lang = "c++"` passed to its constructor. There's probably something wrong at [static/main.js#L505](https://github.com/compiler-explorer/compiler-explorer/blob/main/static/main.js#L505)

### Steps to reproduce

1. Get a compiler for something other than C++
2. Run `make dev EXTRA_ARGS="--language {langId}"`
See that no compiler shows as available

### Expected behavior

The desired lang compilers should show up

### Reproduction link

Not applicable

### Screenshots

Not applicable

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by RubenRBS**
Also adds some more checks for null compilers that got triggered in the
process

Closes #3958

### CODE REVIEW

File: static/main.js [hunk 1]
```diff
@@ -482,6 +482,22 @@ function earlyGetDefaultLangSetting() {
     return Settings.getStoredSettings().defaultLanguage;
 }
 
+function getDefaultLangId(subLangId, options) {
+    var defaultLangId = subLangId;
+    if (!defaultLangId) {
+        var defaultLangSetting = earlyGetDefaultLangSetting();
+        if (defaultLangSetting && options.languages[defaultLangSetting] !== undefined) {
```
Comment (RubenRBS) [inline static/main.js:489]: The only change here is to check that the default language is available. The rest is just code being moved to a function

**Comment by mattgodbolt** [comment]
Oh amazing! I had noticed things be a bit weird here before but was digigng into something else and forgot to come back and see what was going on.

File: static/panes/compiler.js [hunk 2]
```diff
@@ -1411,7 +1411,7 @@ Compiler.prototype.onCompileResponse = function (request, result, cached) {
 Compiler.prototype.postCompilationResult = function (request, result, wasCmake) {
     if (result.popularArguments) {
         this.handlePopularArgumentsResult(result.popularArguments);
-    } else {
+    } else if (this.compiler != null) {
```
Comment (mattgodbolt) [inline static/panes/compiler.js:1414]: mayybe just "this.compiler" to be the opposite of line 2493 (I'm not sure of the subtleties between all the various "nullish or undefinedish" checks)

**Review by mattgodbolt — APPROVED** [review]

**Comment by RubenRBS** [comment]
This is now live

**Comment by RubenRBS** [comment]
This is now live

**Pull request merged**